pip_audit/_dependency_source/requirement.py

```diff
@@ -89,6 +94,20 @@ def collect(self) -> Iterator[Dependency]:
                 reqs: List[InstallRequirement] = []
                 req_names: Set[str] = set()
                 for req in rf.requirements:
+                    if req.req is None:
+                        # For editable requirements that don't have an egg fragment that lists the
```

Comment:
This behaviour is actually slightly different to what `pip-api` was doing. For most editable requirements, `pip-api` does the same thing but it has some special logic for directories where it will look for a `setup.py` on the filesystem and attempt to parse the package name out of it.

Not sure whether we want to replicate this behaviour and whether `pip-requirements-parser` would want this.

Reply:
Maybe raise an issue there? I'd expect identical functionality.

Reply:
> Maybe raise an issue there? I'd expect identical functionality.

Agreed.

Reply:
Opened an issue: https://github.com/nexB/pip-requirements-parser/issues/6.